x5.md

```diff
@@ -8,7 +8,118 @@ created: 2019-04-18
+- client - client (not really ever, this is facilitated through all other connections)
+    - personal chat
+        - ratcheted with X3DH
+    - private group chat
+    - public chat
+- client - mailserver (statusd + ???)
+    - a mailserver identifies itself by ???
+- client - whisper node (statusd)
+    - a node identifies itself by ???
+- client - bootnode (geth)
+    - a bootnode identifies itself by ???
+- client - ENS registry (ethereum blockchain -> default to infura)
+- client - Ethereum RPC (custom geth RPC API -> default to infura API)
+- client - IPFS (Status hosted IPFS gateway -> defaults to ???)
+
+A user in the system is a public-private key pair using the Elliptic-Curve Cryptography secp256k1 that Ethereum uses.
+- A 3-word random name is derived from the public key using the following package
+    - 
+    - This provides an associated human-readble fingerprint to the user's public key
+- A user can optionally add additional layers on top of this keypair
+    - Chosen username
+    - ENS username
+
+All messages sent are encrypted with the public key of the destination and signed by the private key of the given user using the following scheme:
+- private chat
+    - the message is encrypted with the public key of the intended user
+- private group chat
+    - the message is pairwise encrypted with each of the public keys of the members of the private group, e.g. for a group of 6 members, 5 messages will be sent. 
+- public group chat
+    - the message is encrypted with the topic 
+
+## Initial Trust Establishment Specification
+### 1. Contact Discovery
+#### 1.1 Public channels
+- Public group channels in Status are a broadcast/subscription system.  All public messages are encrypted with a symmetric key drived from the channel name, $K_{\text{pub,sym}}$, which is publicly known.
+- A public group channel's symmetric key MUST creation must follow the [web3 API](https://web3js.readthedocs.io/en/1.0/web3-shh.html#generatesymkeyfrompassword)'s `web3.ssh.generateSymKeyFromPassword` function
+- In order to post to a public group channel, a client MUST have a valid account created (as per section [Account Creation Specification](#Account-Creation) TODO: LINK THIS).
+- In order to listen to a public group channel, a client must subscribe to the channel name.  The sender of a message is derived from the message's signature.
+- Discovery of channel names is not currently part of the protocol, and is typically done out of band.  If a channel name is used that has not been used, it will be created.
+- channel name specification:
+    - lower case alphanumeric
+    - no special characters excluding `-`
+    - TODO: other langauge characters
+
+#### 1.2 Private 1:1 messages
+This can be done in a the following ways:
+1. scanning a user generated QR code
+1. discovery through the Status app
+1. asyncronous X3DH key exchange
+1. public key via public channel listening
+    - `status-react/src/status_im/contact_code/core.cljs`
```

Comment:
This will need to be updated, that code is not used anymore, it's already in status-go, and we are in the process of extracting it out, so maybe best to leave a comment

Reply:
kk, will create issue after merge